docs/source/examples/checkpointer.rst

```diff
@@ -0,0 +1,319 @@
+.. _understand_checkpointer:
+
+==============================
+Understanding the Checkpointer
+==============================
+
+This tutorial will walk you through the design and behavior of the checkpointer and associated
+utilities.
+
+
+Overview
+--------
+
+TorchTune checkpointers are designed to be composable components which can be plugged
+into any recipe - training, evaluation or generation. Each checkpointer supports a
+set of models and scenarios making these easy to understand, debug and extend.
+
+TorchTune is designed to be "state-dict invariant".
```

Comment:
Idk if it's too elementary, but could just give a sentence or two on state dicts and how they can differ (different keys, reshaped tensors, etc)